### PR TITLE
[EDM-1539] Replace git-wip

### PR DESCRIPTION
**Pull request by tmilewski**
Currently testing with a wider audience.

`gem install learn-test --pre`
`. learn-test`

Additionally:
- Replaces `require_relative` and `autoload` with `zietwerk`.
- Adds and updates existing tests with Aruba for sandboxed filesystem specs
- Adds Simplecov
- Fixes Rubocop offenses [not all... ;-)]

### CODE REVIEW

File: lib/learn_test/git/wip/base.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                 end
+          else
+            @wip_branch.parent = @working_branch.last_revision
+          end
+
+          new_tree = build_new_tree(@wip_branch.parent)
+          @base.diff(new_tree, @wip_branch.parent)
+
+          # tree_diff = @base.diff(new_tree, @wip_branch.parent)
+          # raise NoChangesError, @wip_branch if tree_diff.count.zero?
```
Comment (drewprice) [inline lib/learn_test/git/wip/base.rb:48]: Does this not need to be here?

**Review by drewprice — APPROVED** [review]
🏆

File: lib/learn_test/reporter.rb [hunk 2]
```diff
@@ -50,16 +47,30 @@ def report
...
+      unless res.success?
+        puts 'There was a problem creating your WIP branch. Not pushing current branch state.'.red if @debug
+        return
+      end
+
+      begin
+        repo.push('origin', "#{res.wip_branch}:refs/heads/fis-wip", { force: true })
+      rescue ::Git::GitExecuteError => e
+        if @debug
+          puts 'There was a problem connecting to Github. Not pushing current branch state.'.red
```
[the remark this answers is not in the document]
Comment (tmilewski) [reply lib/learn_test/reporter.rb:71]: Hahaha! Done!

File: lib/learn_test/git/wip/base.rb (hunk 1 above)
Comment (tmilewski) [reply lib/learn_test/git/wip/base.rb:48]: Keeping it just until wider testing is complete.

File: lib/learn_test/reporter.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (talum) [reply lib/learn_test/reporter.rb]: Omg. Thank you! 🏅

**Pull request merged**